### PR TITLE
[ci] Remove ios32 tests

### PR DESCRIPTION
**Pull request by CaseyHillers**
https://github.com/flutter/flutter/issues/97148

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

### CODE REVIEW

**Review by keyonghan — COMMENTED** [review]
@jmagman @zanderso do we still want to keep those two test dart files? If not, we may also need to clean up the TESTOWNERS.
```
/dev/devicelab/bin/tasks/native_ui_tests_ios.dart
/dev/devicelab/bin/tasks/flutter_gallery__transition_perf_e2e_ios32.dart
```

File: .ci.yaml [hunk 1]
```diff
@@ -109,28 +109,6 @@ platform_properties:
       os: Mac-12.0
       device_os: iOS-15.1
       xcode: 13a233
-  mac_ios32:
```
Comment (godofredoc) [inline .ci.yaml:112]: Do we run these tests on release branches?

Comment (CaseyHillers) [reply .ci.yaml:112]: Yes. We can let this roll through the release process, or we can CP it into the next releases

**Review by jmagman — COMMENTED** [review]
@zanderso should we do this before the `32-bit iOS devices are only supported as part of the “Best Effort” tier` announcement?

https://flutter.dev/go/rfc-32-bit-ios-support#heading=h.8pjtgh69i9ph

**Comment by jmagman** [comment]
> @jmagman @zanderso do we still want to keep those two test dart files? If not, we may also need to clean up the TESTOWNERS.
> 
> ```
> /dev/devicelab/bin/tasks/native_ui_tests_ios.dart
> /dev/devicelab/bin/tasks/flutter_gallery__transition_perf_e2e_ios32.dart
> ```

We should remove `flutter_gallery__transition_perf_e2e_ios32` but `native_ui_tests_ios` is still used in the non-32 bit `Mac_ios native_ui_tests_ios` test.

**Comment by zanderso** [comment]
> @zanderso should we do this before the 32-bit iOS devices are only supported as part of the “Best Effort” tier announcement?

This can happen as soon as we no longer need to test the 2.10 beta/stable release. /cc @godofredoc for when that will be.

**Comment by godofredoc** [comment]
> pen as soon as we no longer need to test the 2.10 beta/stable

We can land this change, next Friday.

**Comment by CaseyHillers** [comment]
Now that 2.10 has shipped, should we remove these tests?

**Review by jmagman — COMMENTED** [review]
This still needs the `TESTOWNERS` change, but other than that @godofredoc says it's good to land, and the change will get cherry-picked in the next hotfix.

**Comment by jmagman** [comment]
> This still needs the `TESTOWNERS` change, but other than that @godofredoc says it's good to land, and the change will get cherry-picked in the next hotfix.

Here:
https://github.com/flutter/flutter/blob/ba01ec8faae9a9be3cd90ee2b7357a803b7a7178/TESTOWNERS#L169-L171

**Review by jmagman — APPROVED** [review]
LGTM!

**Comment by jmagman** [comment]
Google testing looks stuck, there's nothing in FRoB for this PR...

**Comment by CaseyHillers** [comment]
> Google testing looks stuck, there's nothing in FRoB for this PR...

\cc @renyou @angjieli I tried creating it from the FRoB UI, but it says the check already existed for the commit

**Comment by angjieli** [comment]
Feel free to manually merge it. I will take a look at the internal issue.

**Pull request merged**